### ISSUE TITLE
Error initialisng new database

### ISSUE THREAD

**Issue by dstansby** [catalyst-cooperative/pudl#2676]
### Describe the bug
I'm trying to create a new database with the current dev branch (733e90ed8809e53307f2375ddab6838643942f79), and I'm getting an error (which looks like it was introduced by https://github.com/catalyst-cooperative/pudl/pull/2645).

### Bug Severity
How badly is this bug affecting you?
 - **High:** This bug is preventing me from using PUDL.

### To Reproduce
- Checkout current dev branch (33e90ed8809e53307f2375ddab6838643942f79)
-  Run `alembic upgrade head` 

```python
% alembic upgrade head
pudl_settings is being deprecated in favor of environment variables PUDL_OUTPUT and PUDL_INPUT. For more info see: https://catalystcoop-pudl.readthedocs.io/en/dev/dev/dev_setup.html
sqlite and parquet directories are no longer being used. Make sure there is a single directory named 'output' at the root of your workspace. For more info see: https://catalystcoop-pudl.readthedocs.io/en/dev/dev/dev_setup.html
pudl_settings is being deprecated in favor of environment variables PUDL_OUTPUT and PUDL_INPUT. For more info see: https://catalystcoop-pudl.readthedocs.io/en/dev/dev/dev_setup.html
sqlite and parquet directories are no longer being used. Make sure there is a single directory named 'output' at the root of your workspace. For more info see: https://catalystcoop-pudl.readthedocs.io/en/dev/dev/dev_setup.html
INFO  [root] alembic config.sqlalchemy.url: sqlite:////Users/dstansby/pudl/pudl-output/pudl.sqlite
/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py:223: UserWarning: Revision 88d9201ae4c4 referenced from 88d9201ae4c4 -> cd70aa9ce4e9 (head), Remove amount column from retained earnings is not present
  util.warn(
Traceback (most recent call last):
  File "/Users/dstansby/mambaforge/envs/pudl-dev/bin/alembic", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/config.py", line 591, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/config.py", line 585, in main
    self.run_cmd(cfg, options)
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/config.py", line 562, in run_cmd
    fn(
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/command.py", line 378, in upgrade
    script.run_env()
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/base.py", line 576, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/dstansby/github/pudl/migrations/env.py", line 81, in <module>
    run_migrations_online()
  File "/Users/dstansby/github/pudl/migrations/env.py", line 75, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/runtime/environment.py", line 868, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/runtime/migration.py", line 609, in run_migrations
    for step in self._migrations_fn(heads, self):
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/command.py", line 367, in upgrade
    return script._upgrade_revs(revision, rev)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/base.py", line 450, in _upgrade_revs
    for script in reversed(list(revs))
                           ^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 797, in iterate_revisions
    revisions, heads = fn(
                       ^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 1424, in _collect_upgrade_revisions
    for rev in self._parse_upgrade_target(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 1216, in _parse_upgrade_target
    return self.get_revisions(target)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 527, in get_revisions
    resolved_id, branch_label = self._resolve_revision_number(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 751, in _resolve_revision_number
    self._revision_map
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 1113, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
                                           ^^^^^^^^^^^^^^
  File "/Users/dstansby/mambaforge/envs/pudl-dev/lib/python3.11/site-packages/alembic/script/revision.py", line 227, in _revision_map
    down_revision = map_[downrev]
                    ~~~~^^^^^^^^^
KeyError: '88d9201ae4c4'

```

### Expected behavior
A new database.

### Software Environment?
 - Operating System: MacOS 13.4
 - Python version and distribution: Python 3.11.3
 - How did you install PUDL? From source

**Comment by e-belfer** [comment]
Thanks for the write-up of the bug, @dstansby. This issue should have gotten fixed by #2681, but let me know if that's not the case.

**Comment by dstansby** [comment]
👍 looks fixed, thanks a lot!
